libbeat/processors/add_process_metadata/gosiger_cid_provider.go

```diff
@@ -0,0 +1,62 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package add_process_metadata
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/elastic/gosigar/cgroup"
+)
+
+type gosigarCidProvider struct {
+	hostPath     string
+	cgroupPrefix string
+}
+
+func (p gosigarCidProvider) GetCid(pid int) (result string, err error) {
+	cgroups, err := cgroup.ProcessCgroupPaths(p.hostPath, pid)
```

Comment:
I see there is a mock `cidProvider` implementation for testing the processor. But that leaves this part untested. I recommend mocking out the `cgroup.ProcessCgroupPaths` function in order to get test coverage of this provider. `add_docker_metadata` does this with:

https://github.com/elastic/beats/blob/bc71069cf0f77bb8ab1bddc8948f046b9c7d9f78/libbeat/processors/add_docker_metadata/add_docker_metadata.go#L49-L51